src/build.py

```diff
@@ -280,12 +280,19 @@ def Tar(directory, print_content=False):
   return tar
 
 
+def UploadFile(local_name, remote_name):
+  """Archive the file with the given name, and with the LLVM git hash."""
+  if not BUILDBOT_BUILDNUMBER:
+    return
+  buildbot.Link('download', cloud.Upload(local_name, '%s/%s/%s' % (
+      BUILDBOT_BUILDERNAME, BUILDBOT_BUILDNUMBER, remote_name)))
+
+
 def Archive(name, tar):
   """Archive the tar file with the given name, and with the LLVM git hash."""
-  if not BUILDBOT_BUILDERNAME:
```

Comment:
It looks like we have a few places in this file where we test on one or the other of `BUILDBOT_BUILDERNAME` or `BUILDBOT_BUILDNUMBER` to just early-return from a function that does buildbot stuff. Maybe it makes sense to make a global function `IsBuildbot()` and use that? And maybe it can assert that if we think we are a buildbot that we have all the correct info in the env.